### DISCUSSION TITLE
如何发送包含多张图片的聊天记录

### DISCUSSION THREAD

**Discussion by lhqm** [simple-robot/Discussions#161] [Q&A] [answered]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 需求场景
假设用户在向`bot`提起请求后，`bot`需要返回`一系列的消息`,且这些消息`可能在群聊发送`,因此可能`触发群聊发言条数限制`或`造成刷屏`,所以需要封装到`聊天记录`中进行合并发送。
### 预期效果
bot能成功返回聊天记录，且聊天记录内可以包含`单图片消息`、`复杂文本消息`、`合并图片消息`甚至`其他合并的聊天记录`，图片来自`本地图片`、`网络图片`和`用户发送的图片序列化之后得到的ImageId`
### 说明
感谢开发大大的耐心解答，不胜感激！

**Comment by ForteScarlet** [comment, marked as answer]
mirai组件中可以通过 [`MiraiForwardMessageBuilder`](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.message/-mirai-forward-message-builder/index.html) 构建出一个 `MiraiForwardMessage` ，在builder中应该可以填充一系列的消息元素。

也可参考：https://github.com/simple-robot/simbot-component-mirai/pull/62

**Reply by lhqm** [reply]
感谢大大的解答，感激不尽！

**Comment by ilemon996** [comment]
如果想实现消息的发送，简单处理一下文字信息，图片不改变。不是转发消息，是普通的消息发送，应该怎么实现呢？
这是我的 demo，可以发送图片，不过是缩略图不是原图，请问大大应该怎么修改。
    @Listener
    public void getMsgFrom(MiraiGroupMessageEvent event) {
        Group group = event.getGroup();
        MiraiBot bot = event.getBot();

        MiraiReceivedMessageContent msg = event.getMessageContent();
        Messages messages = msg.getMessages();
        MessagesBuilder messagesBuilder = new MessagesBuilder();
        messages.forEach(a -> {
            if(a instanceof MiraiImage image){
                ID id = image.getId();
                messagesBuilder.image(bot, id);
            } else {
                String plainText = msg.getPlainText();
                plainText = plainText + group.getName();
                messagesBuilder.text(plainText);
            }

        });
        group.sendAsync(messagesBuilder.build());
    }
    
    
![image](https://github.com/simple-robot/Discussions/assets/55642977/d4ce3bb7-e9fa-4b9b-80ae-c0335665b8d2)
